src/api_server/src/request/mmds.rs

```diff
@@ -16,9 +17,18 @@ pub fn parse_put_mmds(
 ) -> Result<ParsedRequest, Error> {
     match path_second_token {
         Some(config_path) => match *config_path {
-            "config" => Ok(ParsedRequest::Sync(VmmAction::SetMmdsConfiguration(
-                serde_json::from_slice::<MmdsConfig>(body.raw()).map_err(Error::SerdeJson)?,
-            ))),
+            "config" => {
+                let mmds_config =
+                    serde_json::from_slice::<MmdsConfig>(body.raw()).map_err(Error::SerdeJson)?;
+                match mmds_config.ipv4_addr() {
+                    Some(ipv4_addr) if is_link_local_valid(ipv4_addr) => Ok(ParsedRequest::Sync(VmmAction::SetMmdsConfiguration(mmds_config))),
+                    None => Ok(ParsedRequest::Sync(VmmAction::SetMmdsConfiguration(mmds_config))),
+                    _ => Err(Error::Generic(
+                        StatusCode::BadRequest,
+                        "MMDS configuration `ipv4_address` field must be a valid link-local IPv4 address.\
+                        Please check IETF RFC 3927 for details about the accepted IPv4 address range.".to_string()))
+                }
```

Comment:
Please move this validation where `MmdsConfig` is defined (`vmm_config/mmds.rs`).

The `api_server` should not hold any program logic other than routing API requests.

Reply:
Moved the validation to `MmdsConfig`.